File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        int open = 0;
        
        for(char c : s.toCharArray()) {
            if(c == '(') 
                open++;
            else
                open--;
            
            if(open < 0)
                return false;
        }

        return open == 0;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include <string>
#include <iostream>

using namespace std;

bool solution(string s)
{
    int open = 0;
    
    for(char c : s) {
        if(c == '(') {
            open++;
        } else {
            open--;
        }
        
        if(open < 0) {
            return false;
        }
    }
    
    return open == 0;
}